code/game/objects/structures/traps.dm

```diff
@@ -39,165 +48,172 @@
+		return FALSE
+	return TRUE
+
+///Utility to set the last trigger time,show trap, and deduct charges.
+/obj/structure/trap/proc/post_triggered()
+	flare()
+	lose_charge()
 
-/obj/structure/trap/stun
-	name = "shock trap"
-	desc = ""
-	icon_state = "trap-shock"
-	var/stun_time = 100
-
-/obj/structure/trap/stun/trap_effect(mob/living/L)
-	L.electrocute_act(30, src, flags = SHOCK_NOGLOVES) // electrocute act does a message.
-	L.Paralyze(stun_time)
+///Called when someone steps OFF and we're ready, make sure to call post_triggered()
+/obj/structure/trap/proc/trigger_step_off(mob/living/victim)
+	playsound(src, 'sound/misc/pressurepad_up.ogg', 65, extrarange = 2)
+	return
 
-/obj/structure/trap/stun/hunter
-	name = "bounty trap"
-	desc = ""
-	icon = 'icons/obj/objects.dmi'
-	icon_state = "bounty_trap_on"
-	stun_time = 200
-	sparks = FALSE //the item version gives them off to prevent runtimes (see Destroy())
-	checks_antimagic  = FALSE
-	var/obj/item/bountytrap/stored_item
-	var/caught = FALSE
-
-/obj/structure/trap/stun/hunter/Initialize(mapload)
-	. = ..()
-	time_between_triggers = 10
-	flare_message = "<span class='warning'>[src] snaps shut!</span>"
+///Called when someone steps ON and we're ready, make sure to call post_triggered()
+/obj/structure/trap/proc/trigger_step_on(mob/living/victim)
+	to_chat(victim, span_info("I feel a 'click' beneath me..."))
+	playsound(src, 'sound/misc/pressurepad_down.ogg', 65, extrarange = 2)
+	return
 
-/obj/structure/trap/stun/hunter/Crossed(atom/movable/AM)
-	caught = TRUE
-	. = ..()
+/obj/structure/trap/shock
+	name = "lightning plate trap"
 
-/obj/structure/trap/stun/hunter/flare()
+/obj/structure/trap/shock/trigger_step_on(mob/living/victim)
 	..()
-	stored_item.forceMove(get_turf(src))
-	forceMove(stored_item)
-	if(caught)
-		stored_item.announce_fugitive()
-		caught = FALSE
-
-/obj/item/bountytrap
-	name = "bounty trap"
-	desc = ""
-	icon = 'icons/obj/objects.dmi'
-	icon_state = "bounty_trap_off"
-	var/obj/structure/trap/stun/hunter/stored_trap
-	var/datum/effect_system/spark_spread/spark_system
-
-/obj/item/bountytrap/Initialize(mapload)
-	. = ..()
-	spark_system = new
-	spark_system.set_up(4,1,src)
-	spark_system.attach(src)
-	name = "[name] #[rand(1, 999)]"
-	stored_trap = new(src)
-	stored_trap.name = name
-	stored_trap.stored_item = src
-
-/obj/item/bountytrap/proc/announce_fugitive()
-	spark_system.start()
-	playsound(src, 'sound/blank.ogg', 50, TRUE)
-
-/obj/item/bountytrap/attack_self(mob/living/user)
-	var/turf/T = get_turf(src)
-	if(!user || !user.transferItemToLoc(src, T))//visibly unequips
-		return
-	to_chat(user, "<span class=notice>I set up [src]. Examine while close to disarm it.</span>")
-	stored_trap.forceMove(T)//moves trap to ground
-	forceMove(stored_trap)//moves item into trap
-
-/obj/item/bountytrap/Destroy()
-	qdel(stored_trap)
-	QDEL_NULL(spark_system)
-	. = ..()
+	victim.electrocute_act(30, src, flags = SHOCK_NOGLOVES) // electrocute act does a message.
+	victim.Paralyze(5 SECONDS)
```

Comment:
put the actual stun as a define so mappers can modify this if they want

Reply:
Adding Define